queryserver-orchestrator/src/main/java/org/apache/phoenix/queryserver/orchestrator/QueryServerCanaryOrchestrator.java

```diff
@@ -0,0 +1,117 @@
+package org.apache.phoenix.queryserver.orchestrator;
+
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.BoundedExponentialBackoffRetry;
+
+import org.apache.curator.utils.CloseableUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class QueryServerCanaryOrchestrator {
+	private static final Logger logger = LoggerFactory.getLogger(QueryServerCanaryOrchestrator.class);
+
+	private static String PATH = "/pqs/leader";
+
+	private static Namespace parseArgs(String[] args) {
+
+		ArgumentParser parser = ArgumentParsers.newFor("PQS Canary Orchestrator").build()
+				.description("PQS Canary Orchestrator");
+
+		parser.addArgument("--zkpath", "-zkp").type(String.class).nargs("?").setDefault(PATH)
+				.help("ZKNode path default: " + PATH);
+
+		parser.addArgument("--hostname", "-hn").type(String.class).nargs("?").help("Hostname on "
```

Comment:
Will be good to add defaults here for hostname and port.

Reply:
done